cassandra/connection.py

```diff
@@ -871,3 +892,69 @@ def stop(self):
     def _raise_if_stopped(self):
         if self._shutdown_event.is_set():
             raise self.ShutdownException()
+
+
+class Timer(object):
+
+    cancelled = False
+
+    def __init__(self, timeout, callback):
+        self.end = time.time() + timeout
+        self.callback = callback
+        if timeout < 0:
+            self.on_timeout()
+
+    def __lt__(self, other):
+        return self.end < other.end
+
+    def cancel(self):
+        self.callback = self._noop
+        self.cancelled = True
+
+    def finish(self, time_now):
+        if self.callback is self._noop or time_now >= self.end:
```

Comment:
Good question. Actually this noop concept is residual from an earlier iteration where the time comparison was in a different context. I think I can do away with noop and just use a flag
Will update.